File: app/src/main/java/com/codepath/greghealy/flixster/models/Movie.java
package com.codepath.greghealy.flixster.models;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.parceler.Parcel;

import java.util.ArrayList;
import java.util.List;

@Parcel
public class Movie {
    String backdropPath;
    String posterPath;
    String title;
    String overview;
    Double voteAverage;
    Integer id;

    public Movie() { }

    public Movie(JSONObject jsonObject) throws JSONException {
        backdropPath = "https://image.tmdb.org/t/p/w342" + jsonObject.getString("backdrop_path");
        posterPath = "https://image.tmdb.org/t/p/w342" + jsonObject.getString("poster_path");
        title = jsonObject.getString("title");
        overview = jsonObject.getString("overview");
        voteAverage = jsonObject.getDouble("vote_average");
        id = jsonObject.getInt("id");
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public Integer getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getOverview() {
        return overview;
    }

    public Double getVoteAverage() {
        return voteAverage;
    }

    public static List<Movie> fromJsonArray(JSONArray jsonArray) throws JSONException {
        List<Movie> movies = new ArrayList<>();
        for (int i = 0; i < jsonArray.length(); i++) {
            movies.add(new Movie(jsonArray.getJSONObject(i)));
        }
        return movies;
    }
}

File: app/src/main/java/com/codepath/greghealy/flixster/models/Movie.kt
package com.codepath.greghealy.flixster.models

import org.json.JSONArray
import org.json.JSONObject

class Movie(jsonObject: JSONObject) {
    val backdropPath: String = "https://image.tmdb.org/t/p/w342${jsonObject.getString("backdrop_path")}"
    val posterPath: String = "https://image.tmdb.org/t/p/w342${jsonObject.getString("poster_path")}"
    val title: String = jsonObject.getString("title")
    val overview: String = jsonObject.getString("overview")

    companion object {
        fun fromJsonArray(movieJsonArray: JSONArray) : List<Movie> {
            val movies = ArrayList<Movie>()
            for(i in 0 until movieJsonArray.length()) {
                movies.add(Movie(movieJsonArray.getJSONObject(i)))
            }
            return movies
        }
    }
}